src/osp/Universe.h

```diff
@@ -67,225 +67,89 @@ class Universe
     Universe(Universe &&move) = delete;
     ~Universe() = default;
 
     /**
-     * @return an null satellite
-     */
-    static constexpr Satellite sat_null() { return entt::null; };
-
-    /**
-     * Create a Satellite with default components, and adds it to the universe.
-     * This Satellite will not be assigned a type.
+     * @brief Create a Satellite with default components
+     *
      * @return The new Satellite just created
      */
     Satellite sat_create();
 
     /**
-     * @return Root Satellite of the universe. This will never change.
+     * @return Root Satellite of the universe
      */
     constexpr Satellite sat_root() { return m_root; };
 
     /**
-     * Remove a satellite
+     * @brief Remove a satellite
      */
     void sat_remove(Satellite sat);
 
     /**
-     * Calculate position between two satellites.
-     * @param referenceFrame [in]
-     * @param target [in]
-     * @return relative position of target in SpaceInt vector
+     * @brief Calculate position between two satellites.
+     *
+     * @param referenceFrame [in] Satellite used as reference frame
+     * @param target         [in] Satellite to calculate position to
+     * @return relative position of target in spaceint_t vector
      */
-    Vector3s sat_calc_pos(Satellite referenceFrame, Satellite target) const;
+    Vector3g sat_calc_pos(Satellite referenceFrame, Satellite target) const;
 
     /**
-     * Calculate position between two satellites.
-     * @param referenceFrame [in]
-     * @param target [in]
+     * @brief Calculate position between two satellites.
+     *
+     * @param referenceFrame [in] Satellite used as reference frame
+     * @param target         [in] Satellite to calculate position to
      * @return relative position of target in meters
      */
     Vector3 sat_calc_pos_meters(Satellite referenceFrame, Satellite target) const;
 
     /**
-     * Create a Trajectory, and add it to the universe.
-     * @tparam TRAJECTORY_T Type of Trajectory to construct
-     * @return Reference to new trajectory just created
+     * @brief Ressign indices in the UCompInCoordspace components of satellites
+     *        in a CoordinateSpace's m_toAdd queue
+     *
+     * @param coordSpace [in] Index to CoordinateSpace in m_coordSpaces
      */
-    template<typename TRAJECTORY_T, typename ... ARGS_T>
-    TRAJECTORY_T& trajectory_create(ARGS_T&& ... args);
+    void update_sat_coordspace(uint32_t coordSpace);
 
-    constexpr Registry_t& get_reg() noexcept { return m_registry; }
-    constexpr const Registry_t& get_reg() const noexcept
+    constexpr Reg_t& get_reg() noexcept { return m_registry; }
+    constexpr const Reg_t& get_reg() const noexcept
     { return m_registry; }
 
+    std::vector< std::optional<CoordinateSpace> > m_coordSpaces;
+
 private:
 
     Satellite m_root;
 
-    std::vector<std::unique_ptr<ISystemTrajectory>> m_trajectories;
-
-    Registry_t m_registry;
+    Reg_t m_registry;
 
 };
```

Comment:
Not terribly important, but i find it's frequently very useful for other contributors to a codebase if the end of a class / struct has a comment with it's name.

e.g.

```
}; // class Universe
```